contracts/Registry/StakedRegistry.sol

```diff
@@ -1,40 +1,48 @@
 pragma solidity ^0.4.24;
 
-import './Registry.sol';
+import './OwnedItemRegistry.sol';
+import 'openzeppelin-solidity/contracts/math/SafeMath.sol';
 import 'openzeppelin-solidity/contracts/token/ERC20/ERC20.sol';
 
-contract StakedRegistry is Registry {
-  ERC20 token;
-  uint minStake; // the minimum required amount of tokens staked
-
-  modifier onlyItemOwner(bytes32 id) {
-    require(itemsMetadata[id].owner == msg.sender);
-    _;
-  }
+// handles token stake for owned items. Requires a minimum stake. Allows
+// the owner to increase or decrease stake, as long as it remains above
+// the minimum stake.
+contract StakedRegistry is OwnedItemRegistry {
+  using SafeMath for uint;
 
-  mapping(bytes32 => ItemMetadata) itemsMetadata;
+  ERC20 token;
+  uint minStake;      // minimum required amount of tokens to add an item
 
-  struct ItemMetadata {
-    address owner;
-    uint stakedTokens;
-  }
+  mapping(bytes32 => uint) public ownerStakes;
 
   constructor(ERC20 _token, uint _minStake) public {
     token = _token;
     minStake = _minStake;
   }
 
-  function add(bytes32 data) public returns (bytes32) {
+  function add(bytes32 data) public returns (bytes32 id) {
```

Comment:
One unfortunate drawback is that unlike goldin's implementation, you cannot add a custom amount. With goldin you `apply(item, stakeAmount, data)`. Another argument is that data, which might be important? (people implementing TCR with a UI might have an opinion on how important that data arg is).

If we wanted it to match one way around could be:

```
TokenCuratedRegistry {
  function add(bytes32 item, uint amount, string data) public returns(bytes32 id) {
      id = super.add(item);
      increaseStake(id, stakeAmount - minstake);
      // insert code here to store data
  }
}
```

Thinking out loud...maybe we can make these adjustments as needed later.

Reply:
For the `data` param, maybe we should consider adding this to an extension. Not sure the base class should be opinionated about including a data string.

For the `add()` function with additional stake, we could include a new function, so that `StakedRegistry` still implements the `IRegistry` function `add()`. Maybe something like this:

```
function addWithStake(bytes32 data, uint stakeAmount) public returns(bytes32 id) {
  require(stakeAmount > minStake);
  require(token.transferFrom(msg.sender, this, stakeAmount.sub(minStake)));
  id = add(data);
```